TestingLowerBounds/Renyi.lean

```diff
@@ -51,203 +53,266 @@ lemma exp_mul_llr' [SigmaFinite μ] [SigmaFinite ν] (hμν : μ ≪ ν) :
   have h_pos : 0 < ((∂μ/∂ν) x).toReal :=  ENNReal.toReal_pos hx_pos.ne' hx_lt_top.ne
   rw [← log_rpow h_pos, exp_log (rpow_pos_of_pos h_pos _)]
 
+lemma integrable_rpow_rnDeriv_compProd_right_iff [CountableOrCountablyGenerated α β]
+    (ha_pos : 0 < a) (ha_one : a ≠ 1) (κ η : kernel α β) (μ : Measure α)
+    [IsFiniteKernel κ] [IsFiniteKernel η] [IsFiniteMeasure μ]
+    (h_ac : 1 ≤ a → ∀ᵐ (x : α) ∂μ, κ x ≪ η x) :
+    Integrable (fun x ↦ ((μ ⊗ₘ κ).rnDeriv (μ ⊗ₘ η) x).toReal ^ a) (μ ⊗ₘ η)
+      ↔ (∀ᵐ x ∂μ, Integrable (fun b ↦ ((∂κ x/∂η x) b).toReal ^ a) (η x))
+        ∧ Integrable (fun x ↦ ∫ b, ((∂κ x/∂η x) b).toReal ^ a ∂(η x)) μ := by
+  simp_rw [← integrable_hellingerFun_iff_integrable_rpow ha_one,
+    integrable_f_rnDeriv_compProd_right_iff (stronglyMeasurable_hellingerFun (by linarith))
+    (convexOn_hellingerFun (by linarith))]
+  refine and_congr_right (fun h_int ↦ ?_)
+  rw [← integrable_hellingerDiv_iff h_int h_ac]
+  refine integrable_hellingerDiv_iff' ha_pos ha_one ?_ h_ac
+  simp_rw [← integrable_hellingerFun_iff_integrable_rpow ha_one, h_int]
+
 /-- Rényi divergence of order `a`. If `a = 1`, it is defined as `kl μ ν`, otherwise as
 `(a - 1)⁻¹ * log (ν(α) + (a - 1) * Hₐ(μ, ν))`.
 If `ν` is a probability measure then this becomes the more usual definition
 `(a - 1)⁻¹ * log (1 + (a - 1) * Hₐ(μ, ν))`, but this definition maintains some useful properties
 also for a general finite measure `ν`, in particular the integral form
-`Rₐ(μ, ν) = (a - 1)⁻¹ * log (∫ x, ((∂μ/∂ν) x).toReal ^ a ∂ν)`. -/
+`Rₐ(μ, ν) = (a - 1)⁻¹ * log (∫ x, ((∂μ/∂ν) x) ^ a ∂ν)`. -/
 noncomputable def renyiDiv (a : ℝ) (μ ν : Measure α) : EReal :=
   if a = 1 then kl μ ν
-  else if hellingerDiv a μ ν ≠ ⊤
-    then (a - 1)⁻¹ * log ((ν Set.univ).toReal + (a - 1) * (hellingerDiv a μ ν).toReal)
-    else ⊤
+  else (a - 1)⁻¹ * EReal.log ((↑(ν Set.univ) + (a - 1) * (hellingerDiv a μ ν)).toENNReal)
```

Comment:
Can you please write something in the file docstring (in an "implementation notes" section), or in the docstring of this def, about the use of EReal.log in this definition?